recipes-rxjs/src/app/components/recipe-creation/recipe-creation.component.ts

```diff
@@ -0,0 +1,152 @@
+import { CommonModule } from '@angular/common';
+import {
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  OnInit,
+  ViewChild,
+} from '@angular/core';
+import {
+  FormArray,
+  FormBuilder,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+  AbstractControl,
+  ValidationErrors,
+} from '@angular/forms';
+import { MatCardModule } from '@angular/material/card';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatInputModule } from '@angular/material/input';
+import { MatSelectModule } from '@angular/material/select';
+import { MatButtonModule } from '@angular/material/button';
+import { MatChipsModule, MatChipInputEvent } from '@angular/material/chips';
+import { COMMA, ENTER } from '@angular/cdk/keycodes';
+import { Router } from '@angular/router';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Component({
+  selector: 'app-recipe-creation',
+  standalone: true,
+  imports: [
+    MatFormFieldModule,
+    MatCardModule,
+    MatInputModule,
+    MatIconModule,
+    MatSelectModule,
+    MatButtonModule,
+    MatChipsModule,
+    ReactiveFormsModule,
+    CommonModule,
+  ],
+  templateUrl: './recipe-creation.component.html',
+  styleUrls: ['./recipe-creation.component.css'],
+})
+export class RecipeCreationComponent implements OnInit {
+  form!: FormGroup;
```

Comment:
Define the typed form here (i.e. remove logic from `ngOnInit()`) so that you can access form controls directly.
```suggestion
  form: FormGroup = this.formBuilder.group({
      title: ['', [Validators.required, Validators.maxLength(60)]],
      description: ['', [Validators.required, this.wordCountValidator(25)]],
      image: [null, Validators.required],
      difficulty: [null, Validators.required],
      tags: this.formBuilder.array(
        [],
        [Validators.required, this.minLengthArray(1)],
      ),
    });
```